### PR TITLE
Problem: Tests for different TIPC address types missing

### PR DESCRIPTION
**Pull request by eponsko**
Solution: 
Add tests for binding/connecting with different TIPC address types, and error checks misused addresses.

### CODE REVIEW

**Comment by eponsko** [comment]
Ooops forgot about automake, will have a look at test_hwm_pubsub

**Comment by sigiesec** [comment]
zmq_init is deprecated, you should use zmq_ctx_new instead. I would suggest moving this into setUp and zmq_ctx_term into tearDown,so that every test case has a fresh context (without sockets with pending close e.g.).

Not sure if this resolves your problem on the go. I can have a closer look at the code when back at my desk on Wednesday.

**Comment by sigiesec** [comment]
@eponsko Hm I tried it with setUp/tearDown in branch ZMQers/eponsko-pr2956 and it seems to work: https://travis-ci.org/ZMQers/libzmq/builds/347229430 

Maybe you want to merge the change I added, and then we can merge this PR?

**Comment by eponsko** [comment]
Hi @sigiesec , sorry for lack of action here, been stuck in bed fighting some kind of virus.. Sounds like a good idea, I'll merge later today (hopefully) and the PR should be ready to go!

**Comment by bluca** [comment]
@eponsko is there anything specific that one needs to be able to use TIPC on Linux? I'd like to enable the tests in Travis

**Comment by sigiesec** [comment]
@bluca @eponsko Oh, they aren't enabled on travis? Then I did not test it ;) Forget what I wrote earlier.

**Comment by eponsko** [comment]
@bluca To use TIPC you need to insert the tipc.ko kernel module, unless TIPC support has been compiled into the kernel (unlikely). So I guess automatic testing with Travis will be difficult..

**Comment by bluca** [comment]
What's left to do here?

**Comment by bluca** [comment]
@eponsko I can work on that at some point - there is sudo access on some configs so it might be possible. Anything else before we can merge this?

**Comment by eponsko** [comment]
@bluca Sure that would be nice, it should be enough to just modprobe the tipc module. Normally one would set a node address and enable interfaces as well, but that should not be necessary for these tests. 
I'll update my branch with Simon's patch and add the test to the automake configuration, and then we should be good to go. Sorry again for the delay here, haven't been able to work :/

File: Makefile.am [hunk 1]
```diff
@@ -792,6 +793,9 @@ tests_test_sub_forward_tipc_LDADD = src/libzmq.la
 tests_test_term_endpoint_tipc_SOURCES = tests/test_term_endpoint_tipc.cpp
 tests_test_term_endpoint_tipc_LDADD = src/libzmq.la
 
+tests_test_address_tipc_SOURCES = tests/test_term_endpoint_tipc.cpp
+tests_test_address_tipc_LDADD = src/libzmq.la
```
Comment (bluca) [inline Makefile.am]: needs: ```${UNITY_LIBS}```
and below:
 ```tests_test_address_tipc_CPPFLAGS = ${UNITY_CPPFLAGS}```

**Comment by eponsko** [comment]
@sigiesec I'm having the same issue as before with zmq_ctx_term() and Unity. Even with zmq_ctx_term()  placed in tearDown() the zmq_ctx_term() gets stuck in an infinite poll in signaler::wait(). I'll redo the test without Unity just to verify that it's actually Unity and not something else (I've had issues with the syscall poll() -> tipc_poll() in the kernel module before on certain kernel versions).

**Comment by sigiesec** [comment]
@eponsko Ah great the problem resolved :) I couldn't really have imagined that it could be related to Unity, as it operates on a completely different level.

Could you apply the formatting fixes from https://travis-ci.org/zeromq/libzmq/jobs/349958124#L646 ? Then the PR can be merged :)

**Pull request merged**